packages/cli/core/src/commands/ui/deploy.ts

```diff
@@ -159,14 +160,44 @@ export default class Deploy extends CLICommand {
     this.validateDeployConfigJson(deployConfigPath, deployConfig);
     const hostedPage = this.buildHostedPage(deployConfig);
 
-    if (flags.pageId) {
-      await this.updatePage({...hostedPage, id: flags.pageId});
+    const pageId = flags.pageId || (await this.getPageIdFromPage(hostedPage));
+
+    if (pageId) {
+      await this.updatePage({...hostedPage, id: pageId});
       return;
     }
 
     await this.createPage(hostedPage);
   }
 
+  private async getPageIdFromPage(
```

Comment:
I added my review on this PR: https://github.com/coveo/cli/pull/1235#discussion_r1152126011 instead of this one ;) 

Woops !

Reply:
The continuing discussion here instead of https://github.com/coveo/cli/pull/1235#discussion_r1152126011
-----

I disagree, I think it's better for the 'frontend' code to limit the number of assertions that results in a network egress, especially for non-idempotent calls.
So essentially, I think we should first determine what kind of network call we want to do (update or create) and then have a single call to it.

I agree however that the getPageId is poorly named and could leave to confusion, I'll rework that.

Reply:
addressed in https://github.com/coveo/cli/pull/1234/commits/97116135311dc1c46c86fc8dc9d8bd2c6429e18d

Reply:
Ok but right now, with how the code is written its like this, right ? : 

- If page id is provided -> call updatePage -> exit command

- If page id is not provided 
    - call listPage to check if there is already a page with that name 
    - if yes + they want to overwrite, update page, exit command.
    - if yes + they do not want to overwrite, exit command.
    - if no, create page, exit command.

Or am I just reading it wrong ?
I'm trying to propose to do exactly the same thing, just re-ordered a bit differently in the branching.